Clean sairedis.h header

kcudnik opened the pull request:
* header sairedis.h should be the only header included in OA to give extra enums for sairedis library. No need for expose internal defines to be exposed in this header.
* since SAI already defined bulk APIs for some objects we can start using those directly, no need to expose internal sairedis functions. 
* introduced new sairediscommon.h header for common defines needed by syncd/saiplayer, those defines will later on be moved to "connection" related objects

daall commented on lib/src/sai_redis_route.cpp:
```diff
@@ -376,8 +376,8 @@ const sai_route_api_t redis_route_api = {
-    sai_bulk_remove_route_entry,
-    sai_bulk_set_route_entry_attribute,
-    sai_bulk_get_route_entry_attribute,
+    redis_bulk_create_route_entry,
```
Do you think it would be worth creating some sort of macro like REDIS_GENERIC_BULK_QUAD_API or something similar like we have for the normal APIs?

kcudnik replied:
yes, something similar like this: https://github.com/Azure/sonic-sairedis/blob/master/lib/inc/sai_redis_internal.h#L74 but for bulk, the issue is that validation part of entry needs to be moved sai_meta.cpp like single object validation part, and a extra method for serialization must be selected, and this is a big chang, but at some point i will make this when i will be refactoring meta validation part

daall replied:
Sounds good!

dzhangalibaba commented:
Looks this break vs tests, syncd at vs docker cannot running data as expected .

```
============================================================= test session starts ==============================================================
platform linux2 -- Python 2.7.15+, pytest-3.3.0, py-1.8.0, pluggy-0.6.0 -- /usr/bin/python
cachedir: .cache
rootdir: /home/dzhang/MS_VS/sonic-buildimage/src/sonic-swss/tests, inifile:
collected 1 item                                                                                                                               
test_acl_ctrl.py::TestPortChannelAcl::test_AclCtrl ERROR                                                                                 [100%]
==================================================================== ERRORS ====================================================================
______________________________________________ ERROR at setup of TestPortChannelAcl.test_AclCtrl _______________________________________________
request = <SubRequest 'dvs' for <Function 'test_AclCtrl'>>
    @pytest.yield_fixture(scope="module")
    def dvs(request):
        name = request.config.getoption("--dvsname")
        keeptb = request.config.getoption("--keeptb")
        imgname = request.config.getoption("--imgname")
        fakeplatform = getattr(request.module, "DVS_FAKE_PLATFORM", None)
>       dvs = DockerVirtualSwitch(name, imgname, keeptb, fakeplatform)
conftest.py:825: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
conftest.py:232: in __init__
    self.init_asicdb_validator()
conftest.py:351: in init_asicdb_validator
    self.asicdb = AsicDbValidator(self)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <conftest.AsicDbValidator object at 0x7ff620d2fad0>, dvs = <conftest.DockerVirtualSwitch object at 0x7ff620d138d0>
    def __init__(self, dvs):
        self.adb = swsscommon.DBConnector(1, dvs.redis_sock, 0)
    
        # get default dot1q vlan id
        atbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_VLAN")
    
        keys = atbl.getKeys()
        assert len(keys) == 1
        self.default_vlan_id = keys[0]
    
        # build port oid to front port name mapping
        self.portoidmap = {}
        self.portnamemap = {}
        self.hostifoidmap = {}
        self.hostifnamemap = {}
        atbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_HOSTIF")
        keys = atbl.getKeys()
    
>       assert len(keys) == 32
E       assert 0 == 32
E        +  where 0 = len(())
conftest.py:48: AssertionError



------------------------------------------------------------ Captured stdout setup -------------------------------------------------------------
Actual changes:
tx-checksumming: off
```

kcudnik commented:
how i can run those tests locally ?

kcudnik commented:
im testing this change on https://github.com/Azure/sonic-buildimage/pull/3872 this pull request, im using it to just test, moving sairedis pointer to https://github.com/Azure/sonic-sairedis/commit/3fbd0c5ecb426cdd9fd2811452d05ade32fa4389 fails the build, but errors i get are differnt than yours: https://sonic-jenkins.westus2.cloudapp.azure.com/job/vs/job/buildimage-vs-all-pr/1791/console 

not entirely sure what happened there, and not sure if this is caused by my change or some other unrelated change or build system issue, i moved 1 commit behind and running it right now will update after it finish, if it fails i will also try to bring sairedis pointer to where it is at current to see if the one that is set on 

```
> request = <SubRequest 'dvs' for <Function 'test_InvalidNexthop'>>

    @pytest.yield_fixture(scope="module")
    def dvs(request):
        name = request.config.getoption("--dvsname")
        keeptb = request.config.getoption("--keeptb")
        imgname = request.config.getoption("--imgname")
        fakeplatform = getattr(request.module, "DVS_FAKE_PLATFORM", None)
>       dvs = DockerVirtualSwitch(name, imgname, keeptb, fakeplatform)

conftest.py:826: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
conftest.py:231: in __init__
    self.check_ready()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <conftest.DockerVirtualSwitch object at 0x7f7cf3646650>, timeout = 30

    def check_ready(self, timeout=30):
        '''check if all processes in the dvs is ready'''
    
        re_space = re.compile('\s+')
        process_status = {}
        ready = False
        started = 0
        while True:
            # get process status
            res = self.ctn.exec_run("supervisorctl status")
            try:
                out = res.output
            except AttributeError:
                out = res
            for l in out.split('\n'):
                fds = re_space.split(l)
                if len(fds) < 2:
                    continue
                process_status[fds[0]] = fds[1]
    
            # check if all processes are running
            ready = True
            for pname in self.alld:
                try:
                    if process_status[pname] != "RUNNING":
                        ready = False
                except KeyError:
                    ready = False
    
            # check if start.sh exited
            if process_status["start.sh"] != "EXITED":
                ready = False
    
            if ready == True:
                break
    
            started += 1
            if started > timeout:
>               raise ValueError(out)
E               ValueError: arp_update                       STOPPED   Not started
E               bgpd                             STOPPED   Not started
E               buffermgrd                       RUNNING   pid 443, uptime 0:00:32
E               fpmsyncd                         RUNNING   pid 351, uptime 0:00:44
E               intfmgrd                         RUNNING   pid 396, uptime 0:00:38
E               nbrmgrd                          RUNNING   pid 451, uptime 0:00:31
E               neighsyncd                       RUNNING   pid 334, uptime 0:00:48
E               orchagent                        EXITED    Dec 11 05:45 PM
E               portmgrd                         RUNNING   pid 386, uptime 0:00:40
E               portsyncd                        RUNNING   pid 227, uptime 0:00:49
E               redis-server                     RUNNING   pid 135, uptime 0:00:53
E               restore_neighbors                STOPPED   Not started
E               rsyslogd                         RUNNING   pid 107, uptime 0:00:55
E               sflowmgrd                        RUNNING   pid 470, uptime 0:00:28
E               start.sh                         EXITED    Dec 11 05:45 PM
E               staticd                          RUNNING   pid 433, uptime 0:00:34
E               syncd                            RUNNING   pid 147, uptime 0:00:52
E               teammgrd                         RUNNING   pid 359, uptime 0:00:43
E               teamsyncd                        RUNNING   pid 343, uptime 0:00:46
E               vlanmgrd                         RUNNING   pid 404, uptime 0:00:37
E               vrfmgrd                          RUNNING   pid 367, uptime 0:00:41
E               vxlanmgrd                        RUNNING   pid 459, uptime 0:00:29
E               zebra                            RUNNING   pid 422, uptime 0:00:35
```

dzhangalibaba commented:
Hi Kamil, I suggest you pull master on local and update sairedis head to this PR's commit.
you can follow https://github.com/Azure/sonic-swss/tree/master/tests

Jenkins always give some weird error which is not stable.

Yesterday I did a fresh pull , and update sairedis header to latest, failed. then I tried updated to this PR, failed again. Then I tried update to the previous commit. It passed all.

kcudnik commented:
Hey, found issue, it's here: https://github.com/Azure/sonic-sairedis/pull/543/files#diff-91b3e380b7f6f4a0a048ca8296a12e0dL3105
and as mentioned before it was fixed here: https://github.com/Azure/sonic-sairedis/pull/548
this is when orchagent query getSupportedDebugCounterAmounts, and it fails since syncd doesn't properly responds, and OA waits for 60 seconds and don't get any response and fails

can you pull master on sairedis and see if it passes for you ? i will also post check on sonic-buildimage

dzhangalibaba commented:
> Hey, found issue, it's here: https://github.com/Azure/sonic-sairedis/pull/543/files#diff-91b3e380b7f6f4a0a048ca8296a12e0dL3105
> and as mentioned before it was fixed here: #548
> this is when orchagent query getSupportedDebugCounterAmounts, and it fails since syncd doesn't properly responds, and OA waits for 60 seconds and don't get any response and fails
> 
> can you pull master on sairedis and see if it passes for you ? i will also post check on sonic-buildimage
> 
> yea build passed from master including this fix: [Azure/sonic-buildimage#3872](https://github.com/Azure/sonic-buildimage/pull/3872) (look at build status vs)

Hi Kamil, looks good now when I update to the fix commit, all passed
